src/factories/fardetectors/FarDetectorLinearTracking_factoryT.h

```diff
@@ -0,0 +1,100 @@
+// Created by Simon Gardner
+// Subject to the terms in the LICENSE file found in the top-level directory.
+//
+
+#pragma once
+
+// Event Model related classes
+#include <edm4eic/TrackParametersCollection.h>
+#include <algorithms/fardetectors/FarDetectorLinearTracking.h>
+
+#include <extensions/jana/JChainMultifactoryT.h>
+#include <extensions/spdlog/SpdlogMixin.h>
+#include <spdlog/logger.h>
+
+namespace eicrecon {
+
+    class FarDetectorLinearTracking_factoryT :
+    public JChainMultifactoryT<FarDetectorLinearTrackingConfig>,
+    public SpdlogMixin {
+
+    public:
+
+      explicit FarDetectorLinearTracking_factoryT(
+          std::string tag,
+          const std::vector<std::string>& input_tags,
+          const std::vector<std::string>& output_tags,
+          FarDetectorLinearTrackingConfig cfg):
+          JChainMultifactoryT(std::move(tag), input_tags, output_tags, cfg) {
+
+          DeclarePodioOutput<edm4eic::TrackParameters>(GetOutputTags()[0]);
+      }
+
+        /** One time initialization **/
+        void Init() override {
+          auto app = GetApplication();
+
+          // SpdlogMixin logger initialization, sets m_log
+          InitLogger(app, GetPrefix(), "info");
+
+          auto cfg = GetDefaultConfig();
+
+          m_geoSvc = app->GetService<JDD4hep_service>();
+
+          if (cfg.detconf.readout.empty()) {
+            throw JException("Readout is empty");
+          }
+
+          try {
+            id_dec = m_geoSvc->detector()->readout(cfg.detconf.readout).idSpec().decoder();
+            if (!cfg.detconf.moduleField.empty()) {
+              cfg.detconf.module_idx = id_dec->index(cfg.detconf.moduleField);
+              logger()->debug("Find module field {}, index = {}", cfg.detconf.moduleField, cfg.detconf.module_idx);
+            }
+            if (!cfg.detconf.layerField.empty()) {
+              cfg.detconf.layer_idx = id_dec->index(cfg.detconf.layerField);
+              logger()->debug("Find layer field {}, index = {}", cfg.detconf.layerField, cfg.detconf.layer_idx);
+            }
+            if (!cfg.detconf.xField.empty()) {
+              cfg.detconf.x_idx = id_dec->index(cfg.detconf.xField);
+              logger()->debug("Find layer field {}, index = {}",  cfg.detconf.xField, cfg.detconf.x_idx);
+            }
+            if (!cfg.detconf.yField.empty()) {
+              cfg.detconf.y_idx = id_dec->index(cfg.detconf.yField);
+              logger()->debug("Find layer field {}, index = {}", cfg.detconf.yField, cfg.detconf.y_idx);
+            }
+          } catch (...) {
+            logger()->error("Failed to load ID decoder for {}", cfg.detconf.readout);
+            throw JException("Failed to load ID decoder");
+          }
```

Comment:
We should not modify `cfg` in the factory. Assume it will become `const` at some point, at least within the scope of this class. This must be done in the algorithm.